lib/runner.js

```diff
@@ -192,16 +192,19 @@ module.exports = {
     var tracker = new Tracker();
     var finish = createFinish(shOpts, opts, tracker, callback);
 
-    // Get mapper (queue vs. non-queue)
     var queue = opts.queue;
-    var map = queue ?
-      async.mapLimit.bind(async, cmds, queue) :
-      async.map.bind(async, cmds);
+    var bail = opts.bail;
+    var errors = [];
 
     log.info("concurrent", "Starting with queue size: " + chalk.magenta(queue || "unlimited"));
-    map(function (cmd, cb) {
-      retry(cmd, shOpts, _.extend({ tracker: tracker }, opts), cb);
-    }, finish);
+    async.mapLimit(cmds, queue || Infinity, function (cmd, cb) {
+      retry(cmd, shOpts, _.extend({ tracker: tracker }, opts), function (err) {
+        errors.push(err);
+        cb(bail ? err : null);
+      });
+    }, function () {
+      finish(errors.length ? errors[0] : null);
```

Comment:
Only returns the first error. I kinda want to report them all, but returning the array causes `err.message` printing to fail. Could combine them into one error somehow...